src/ros/laser_scan_observer.h

```diff
@@ -8,13 +9,22 @@
 #include "../core/sensor_data.h"
 #include "topic_with_transform.h"
 
+/**
+ * \brief This class observes for LaserScan and stores information about location.
+ */
+
 class LaserScanObserver : public TopicObserver<sensor_msgs::LaserScan> {
   using ScanPtr = boost::shared_ptr<sensor_msgs::LaserScan>;
 public: //methods
 
   LaserScanObserver(bool skip_max_vals = false):
     _skip_max_vals(skip_max_vals),
-    _prev_x(0), _prev_y(0), _prev_yaw(0) {}
+    _prev_x(0), _prev_y(0), _prev_yaw(0) {} /// < Constructor of this class
+/**
+ * Converts Transform and LaserScan after robot has moved.
```

Comment:
"Converts ROS-specific structures that hold sensor data to internal framework's structures. Laser scan filtering is performed as part of the conversion."